redev_exclusive_scan.h

```diff
@@ -0,0 +1,18 @@
+#ifndef REDEV_EXCLUSIVE_SCAN_H
+#define REDEV_EXCLUSIVE_SCAN_H
+namespace redev {
+  template <typename InputIt, typename OutputIt, typename T>
+  OutputIt exclusive_scan(InputIt first, InputIt last, OutputIt d_first, T init) {
```

Comment:
NVHPC/pgi compilers don't provide `std::exclusive_scan`?

Reply:
I'm using `NVHPC 21.7` on summit which has the corresponding XGC build working: https://xgc.pppl.gov/html/cmake_build_instructions.html

It doesn't seem to have inclusive/exclusive scan.